### PR TITLE
Update CP2K recipe for AOCC compiler

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
Changes include updates in `CP2K` recipe for `AOCC` compiler
- `libint` version dependency
- Updates for `AOCC-5.0` compiler

### CODE REVIEW

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -304,6 +309,8 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
     with when("@2022: +rocm"):
         depends_on("hipblas")
         depends_on("hipfft")
+    # The CMake build system and AOCC are not compatible as of AOCC 5
+    requires("build_system=makefile", when="%aocc")
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/cp2k/package.py:310]:
```suggestion

    # The CMake build system and AOCC are not compatible as of AOCC 5
    requires("build_system=makefile", when="%aocc")
```

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 2]
```diff
@@ -192,6 +192,11 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
                 "libint@2.6.0:+fortran tune=cp2k-lmax-{0}".format(lmax),
                 when="@7.0: lmax={0}".format(lmax),
             )
+            # AOCC only works with libint@2.6.0
+            depends_on(
+                "libint@=2.6.0+fortran tune=cp2k-lmax-{0}".format(lmax),
+                when="@7.0: lmax={0} %aocc".format(lmax),
+            )
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/cp2k/package.py]:
```suggestion
            # AOCC only works with libint@2.6.0
            depends_on(
                f"libint@=2.6.0+fortran tune=cp2k-lmax-{lmax}",
                when=f"@7.0: lmax={lmax} %aocc",
            )
```
I think it is recommended to use f-strings now.

**Review by RMeli — COMMENTED** [review]
A few style suggestions, otherwise LGTM (but I haven't tested the build myself).

File: var/spack/repos/builtin/packages/cp2k/package.py (hunk 2 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/cp2k/package.py]: Thanks
changes pushed as suggested

File: var/spack/repos/builtin/packages/cp2k/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/cp2k/package.py:310]: Thanks,
Changes pushed as suggested.

**Review by RMeli — APPROVED** [review]

**Comment by RMeli** [comment]
@VivekAMD do you want your account to be indicated as co-author in the squashed commit?

**Comment by VivekAMD** [comment]
> @VivekAMD do you want your account to be indicated as co-author in the squashed commit?

Thank you @RMeli, we can go without having my account as co-author.

**Pull request merged**